redriver/src/main/java/com/flipkart/flux/redriver/scheduler/MessageScheduler.java

```diff
@@ -68,17 +67,12 @@ public void addMessage(ScheduledMessage scheduledMessage) {
     }
 
     public void removeMessage(Long taskId) {
-        final Optional<ScheduledMessage> scheduledMessageOptional
-                = this.messages.stream().filter((m) -> m.getTaskId().equals(taskId)).findFirst();
-        if (scheduledMessageOptional.isPresent()) {
-            final ScheduledMessage message = scheduledMessageOptional.get();
-            /* It is important that we delete from priority queue first. Its okay even if we the schedule for removal call fails.
-                At most we will send a message that was not supposed to be sent but the receiver should be able to handle
-                such cases by consulting its own DB.
-             */
-            this.messages.remove(message);
-            this.messageManagerService.scheduleForRemoval(message);
-        }
+        /* It is important that we delete from priority queue first. Its okay even if we the schedule for removal call fails.
+            At most we will send a message that was not supposed to be sent but the receiver should be able to handle
+            such cases by consulting its own DB.
+        */
+        this.messages.removeIf((m) -> m.getTaskId().equals(taskId));
```

Comment:
Hey, can we use linkedHashSet instead of PriorityQueue. If tasks are registered roughly in the order of their executionTime then the task with the earliest execTime should be at the front of linkedList.